Client/Frontend/Browser/TabManager.swift

```diff
@@ -244,8 +244,6 @@ class TabManager: NSObject {
             selectedTab?.createWebview()
             restoreTab(t)
         }
-
-        assert(tab === selectedTab, "Expected tab is selected")
         
```

Comment:
Can we replace this with a check and `log.error` + some info like the current tab, what the selected one is, etc. in case we notice it happen locally

Reply:
Requested change at https://github.com/brave/brave-ios/pull/3212/commits/6a6d16f0075982a5f5dcda2d86333c13fa81bd2d